tests/basics/try_finally_break.py

```diff
@@ -0,0 +1,76 @@
+# test break within (nested) finally
+
+# basic case with break in finally
+def f():
+    for _ in range(2):
+        print(1)
+        try:
+            pass
+        finally:
+            print(2)
+            break
+            print(3)
+        print(4)
+    print(5)
+f()
+
+# where the finally swallows an exception
+def f():
+    lst = [1, 2, 3]
+    for x in lst:
+        print('a', x)
+        try:
+            raise Exception
+        finally:
+            print(1)
+            break
+        print('b', x)
+f()
+
+# basic nested finally with break in inner finally
+def f():
+    for i in range(2):
+        print('iter', i)
+        try:
+            raise TypeError
+        finally:
+            print(1)
+            try:
+                raise ValueError
+            finally:
+                break
+print(f())
+
+# similar to above but more nesting
+def f():
+    for i in range(2):
+        try:
+            raise ValueError
+        finally:
+            print(1)
+            try:
+                raise TypeError
+            finally:
+                print(2)
+                try:
+                    pass
+                finally:
+                    break
+print(f())
+
+# lots of nesting
+def f():
+    for i in range(2):
+        try:
+            raise ValueError
+        finally:
+            print(1)
+            try:
+                raise TypeError
+            finally:
+                print(2)
+                try:
+                    raise Exception
+                finally:
+                    break
```

Comment:
Wouldn't it be interesting to add here more "finally" configuration cases, e.g. mix of where sub-finally is nested within parent's "finally" vs parent's body?

Reply:
We also don't have a single test for try/except/else in the testsuite.

Reply:
Yes it would be good to add both these kinds of tests.  The former can go in this set of patches here.  The latter (else) would be a separate thing.

Reply:
Bottom line, is that merging this without testing the `else:` clauses shouldn't be done to avoid surprises.